### PR TITLE
mosquitto_sub: add page

### PR DESCRIPTION
**Pull request by mmphego**
- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform folder (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/master/CONTRIBUTING.md#guidelines).

### CODE REVIEW

File: pages/common/mosquitto_sub.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# mosquitto_sub
+
+> A simple MQTT version 3.1.1 client that will subscribe to topics and print the messages that it receives.
+
+- Subscribe to all broker status messages:
+
+`mosquitto_sub -v -t {{\$SYS/#}}`
```
Comment (sbrl) [inline pages/common/mosquitto_sub.md]: Are there any long forms of these options available? In tldr-pages, we prefer to use the long options where possible, as they are more descriptive.

File: pages/common/mosquitto_sub.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+- Subscribe to "topic" sensors/temperature information on `localhost` with `QoS` 1:
```
Comment (sbrl) [inline pages/common/mosquitto_sub.md]: Perhaps this will read better?

```suggestion
- Subscribe to the topic `sensors/temperature` information on `localhost` with `QoS` 1:
```

Also, what does `QoS` stand for? Could we expand it at all if it's an abbreviation?

File: pages/common/mosquitto_sub.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+`mosquitto_sub -t {{sensors/temperature}} -q {{1}}`
+
+- Subscribe to temperature updates on multiple machines. This expects each machine to be publishing its temperature to "sensors/machines/`HOSTNAME`/temperature/`HD_NAME`":
```
Comment (sbrl) [inline pages/common/mosquitto_sub.md]: What's `HD_NAME`?

File: pages/common/mosquitto_sub.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+`mosquitto_sub -t {{sensors/machines/+/temperature/+}}`
+
+- Subscribe to all topics and `QoS` 2, and print to STDOUT in colour where supported:
```
Comment (sbrl) [inline pages/common/mosquitto_sub.md]: I'm unsure that `STDOUT` needs to be in capital letters here.

File: pages/common/mosquitto_sub.md (hunk 2 above)
Comment (mmphego) [reply pages/common/mosquitto_sub.md]: `QoS` stands for Quality of Service.
Will update the doc

File: pages/common/mosquitto_sub.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+`mosquitto_sub -F {{'\e[92m%t \e[96m%p\e[0m'}} -q {{2}} -t {{'#'}}`
```
[the remark this answers is not in the document]
Comment (mmphego) [reply pages/common/mosquitto_sub.md]: Removed as it seems unnecessary.

File: pages/common/mosquitto_sub.md (hunk 3 above)
Comment (mmphego) [reply pages/common/mosquitto_sub.md]: `HD_NAME` relates to Hard Disk x; where `x` is the name of the harddisk

File: pages/common/mosquitto_sub.md (hunk 4 above)
Comment (mmphego) [reply pages/common/mosquitto_sub.md]: fixed

File: pages/common/mosquitto_sub.md (hunk 1 above)
Comment (mmphego) [reply pages/common/mosquitto_sub.md]: Updated

File: pages/common/mosquitto_sub.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+- Subscribe to all broker status messages publishing on `iot.eclipse.org` port 1885 and print published messages verbosely, by default hostname resolves to `localhost` and port `1883`:
+
+`mosquitto_sub -v -h "iot.eclipse.org" -p 1885 -t {{\$SYS/#}}`
+
+- Subscribe to the topic `sensors/temperature` information on `localhost` with Quality of Service (`QoS`) set to 1:
+
+`mosquitto_sub -t {{sensors/temperature}} -q {{1}}`
+
+- Subscribe to Hard disk (HD_NAME) temperature updates on multiple machines defined by `HOSTNAME`. This expects each machine to be publishing it's temperature to "sensors/machines/`HOSTNAME`/temperature/HD_NAME":
```
Comment (agnivade) [inline pages/common/mosquitto_sub.md]: I don't think this needs to be Hard disk temperatures, or the middle metric has to be hostname. The + is just a catchall for whatever metric comes in between. (Please correct me if I am wrong)

We can shorten this to - `Subscribe to multiple topics matching a given pattern. (+ takes any metric name):`

File: pages/common/mosquitto_sub.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
+# mosquitto_sub
+
+> A simple MQTT version 3.1.1 client that will subscribe to topics and print the messages that it receives.
+
+- Subscribe to all broker status messages publishing on `iot.eclipse.org` port 1885 and print published messages verbosely, by default hostname resolves to `localhost` and port `1883`:
+
+`mosquitto_sub -v -h "iot.eclipse.org" -p 1885 -t {{\$SYS/#}}`
+
+- Subscribe to the topic `sensors/temperature` information on `localhost` with Quality of Service (`QoS`) set to 1:
```
Comment (agnivade) [inline pages/common/mosquitto_sub.md]: I think this can be the first example, since this is short. And also mention the default port of 1883 here, so that you don't have to repeat that info in the above example.

**Comment by mmphego** [comment]
PR is ready to be merged.

Regards,
Mpho

Sent via Android device

**Comment by mmphego** [comment]
Thank you, will do.
